features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/api.js

```diff
@@ -2026,14 +2026,31 @@ class API extends Resource {
             return client.apis.APIs.getAPI({ apiId: id }, this._requestMetaData());
         });
         if (callback) {
-            console.log('original object from API1');
             return promiseGet.then(callback);
         } else {
-            console.log('original object from API2' + promiseGet.obj);
             return promiseGet;
         }
     }
 
+    /**
+     * Generate Internal Key
+     * @param id {string} UUID of the api.
+     * @param callback {function} A callback function to invoke after receiving successful response.
+     * @returns {promise} With given callback attached to the success chain else API invoke promise.
+     */
+    static generateInternalKey(id, callback = null) {
+        const apiClient = new APIClientFactory().getAPIClient(Utils.getCurrentEnvironment(), Utils.CONST.API_CLIENT).client;
+        const promiseKey = apiClient.then((client) => {
+            return client.apis.APIs.generateInternalAPIKey({ apiId: id }, this._requestMetaData());
+        });
+        if (callback) {
+            return promiseKey.then(callback);
```

Comment:
This callback logic is actually redundant, Shall we remove this along with the method argument